### PR TITLE
Add react support (jsx and tsx)

### PR DESCRIPTION
**Pull request by fouteox**
Hey, thanks for this package.

Here is support for react and inertia.

I included two versions, a classic jsx and a typescript.

### CODE REVIEW

File: resources/ts/LoginLink.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    className?: string;
+    email?: string;
+    guard?: string;
+    keyId?: string;
+    label?: string;
+    redirectUrl?: string;
+    userAttributes?: Record<string, any>;
+}
+
+const LoginLink: React.FC<LoginLinkProps> = ({
```
Comment (sebastiandedeyne) [inline resources/ts/LoginLink.tsx:14]:
```suggestion
export default function LoginLink({ … }: LoginLinkProps) {
```

File: resources/js/LoginLink.jsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { router } from '@inertiajs/react'
+
+const LoginLink = ({
```
Comment (sebastiandedeyne) [inline resources/js/LoginLink.jsx:4]: For consistency with our other projects and packages, could you use a function component instead?

```tsx
export default function LoginLink
```

File: README.md [hunk 3]
```diff
@@ -190,21 +190,43 @@ public function share(Request $request): array
...
+    <LoginLink />
+)}
+
+// or
+
+{page.props.environment === 'local' && (
+    <LoginLink
+        label="Login as user@example.com"
+        className="pb-3 text-red-500"
+        redirect-url="route('dashboard')"
```
Comment (sebastiandedeyne) [inline README.md:222]:
```suggestion
        redirectUrl={route('dashboard')}
```

File: README.md [hunk 4]
```diff
@@ -175,9 +175,9 @@ If the user that needs to be logged in does not exist, the package will use the
 
 If you don't want this behaviour, set `automatically_create_missing_users` in the `local-link` config file to `false`.
 
-### Usage with Vue and InertiaJS
+### Usage with Vue or React and InertiaJS
```
Comment (sebastiandedeyne) [inline README.md:178]:
```suggestion
### Usage with Vue or React and Inertia.js
```

Could you clean up other usages in the readme too? 🙏

File: resources/ts/LoginLink.tsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    className = 'underline',
+    email = null,
+    guard = null,
+    keyId = null,
+    label = 'Login',
+    redirectUrl = null,
+    userAttributes = null,
+}) => {
+    const submit = (event: FormEvent) => {
```
Comment (sebastiandedeyne) [inline resources/ts/LoginLink.tsx:23]:
```suggestion
    function handleSubmit(event: FormEvent) {
```

**Review by sebastiandedeyne — CHANGES_REQUESTED** [review]
Thanks for your PR! I noted some small code style issues before merging.

**Comment by fouteox** [comment]
> Thanks for your PR! I noted some small code style issues before merging.

done!

**Review by sebastiandedeyne — APPROVED** [review]

**Pull request merged**